### DISCUSSION TITLE
Unable to get ESP32 Audio Kit working with either AUDIOKIT_BOARD value

### DISCUSSION THREAD

**Discussion by IrregularShed** [pschatzmann/arduino-audio-tools#121] [Q&A] [answered]
Hi Phil (and anyone else reading).

I've been trying to work my way out of the problems caused by there being two different ESP32 Audio Kit v2.2 boards. I'm a bit stuck though and hope there's a suggestion of where to go next.

I've opened the example `streams-synth-audiokit` and when I first compiled it I received the `Error: ->Wire.endTransmission: 2` response, with `please verify your AUDIOKIT_BOARD setting: 5`. This lead me to discover that there were two different DACs, so I added `#define AUDIOKIT_BOARD 6` at the top of the sketch to use the AC101 code. Now, I don't get the `Wire.endTransmission` output, but I have a different error echoed out:

```
[W] AudioKit.h : 43 - sd_active = false
Error:   reset failed!
Error:   codec init failed!
Error:   audio_hal_init
[E] AudioKit.h : 185 - begin faild: please verify your AUDIOKIT_BOARD setting: 6
```

So from this I can conclude my board has the older sound chip, and that AudioKit has detected it on the I2C bus, but I clearly need to do something else for reset and codec initialisation to succeed. Any suggestions? By the way, I'm using the library that I cloned from git this morning, and in Arduino I've got the board set to ESP32 Dev Module with PSRAM enabled. DIP switches 2 and 3 are on, the rest are off.

**Comment by pschatzmann** [comment]
Please read the readme of the [AudioKit](https://github.com/pschatzmann/arduino-audiokit) project: You need to update the setting in the configuration file as indicated there....

**Reply by IrregularShed** [reply]
I'd already read through that but I was having difficulty following - I couldn't find an existing `AudioKitSettings.h` file in this project and none of the examples in the AudioKit project have one of those files either, so I don't know where that file would go. I thought having `#define AUDIOKIT_BOARD 6` at the very top of the .ino file would be enough, and the fact the serial output has changed suggests that it has picked it up, but I'm obviously missing something.

**Comment by pschatzmann** [comment]
if you check in Arduino/libraries/arduino-audiokit/src directory you should find the AudioKitSettings.h file
Unfortunately adding it in the sketch is not good enough because not all functionality will see this...

**Reply by IrregularShed** [reply]
Still nothing 😞 I've modified the `AudioKitSettings.h` file I found, inserting the `#define` immediately after `#pragma once` line. I also restarted the Arduino IDE in case there was something cached somewhere, and tried a different example for the same reason. I'm still getting the same output:

```
rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1324
ho 0 tail 12 room 4
load:0x40078000,len:13508
load:0x40080400,len:3604
entry 0x400805f0
[W] AudioKit.h : 43 - sd_active = false
Error:   reset failed!
Error:   codec init failed!
Error:   audio_hal_init
[E] AudioKit.h : 185 - begin faild: please verify your AUDIOKIT_BOARD setting: 6
```

![image](https://user-images.githubusercontent.com/190127/157300436-92b5285c-a0ef-4707-abe5-1cc530a27f5e.png)

**Comment by pschatzmann** [comment]
Hmm, that's strange and at the moment I have no idea what your issue could be. I re-tested on my side the two different boards and for me everything is working.

Did you ever manage to have your board working with any other Software (e.g. ADF) ?

What's the exact Version Number on your board. On my boards I have V2.2 3478 for the ES8388 and V2.2 2762 for the AC101

The only thing that you could try is to set AUDIOKIT_USE_WIRE to 0 to use the native ESP32 I2C implementation. But my hopes are low...

**Comment by IrregularShed** [comment]
I honestly can't remember if I've ever had anything running on this. I got it shortly before contracting Covid and about 5 months of my memory has gone. (Speaking of lost memory, I've also completely erased the flash on the board using esptool.py, but that was just grasping at straws.)

The version is V2.2 2957, and setting AUDIOKIT_USE_WIRE didn't make any difference either 😕 I've got a LyraT board here as well, so I'll try getting an example onto that later on to make sure that my PC is building things correctly. (Thanks for your help so far!)

**Comment by pschatzmann** [comment]
I googled your version and it might be that on that revisions pins are different from my revision...
Try with setting 5
and in src/audio_board/ai_thinker_v2_2/board_pins_config.c change
from
        i2c_config->sda_io_num = GPIO_NUM_33; // GPIO_NUM_18
        i2c_config->scl_io_num = GPIO_NUM_32; // GPIO_NUM_23
to
        i2c_config->sda_io_num =  GPIO_NUM_18
        i2c_config->scl_io_num =  GPIO_NUM_23

**Reply by IrregularShed** [reply]
Getting closer! It compiles and runs, and the output in the serial monitor shows my KeyOn/KeyOff events as I press the buttons. However, there's no sound at all, on either the headphone jack or the speaker connectors. The trace on my oscilloscope was completely flat, no electrical noise at all. I used the same example as I built for my LyraT board, which worked correctly.

I googled as well, and judging by [this issue](https://github.com/Ai-Thinker-Open/ESP32-A1S-AudioKit/issues/26) they really messed up on this version.

**Comment by pschatzmann** [comment]
Looking further in the linked issue change
        i2s_config->bck_io_num = GPIO_NUM_27;
to 
        i2s_config->bck_io_num = GPIO_NUM_5;

**Reply by IrregularShed** [reply]
That's got it! Going back to `streams-generator-audiokit` I can hear the tone now. Initially all I could hear was crackling noise, but I realised that the logger was sending a lot of data over the serial port; with it changed from info to warning the crackle had gone. The tone is very quiet, but it's definitely there!

However... there's still an issue, out of our hands. IO5 is also KEY6! This meant that examples using the keys keep registering KEY6 being pressed and released as the bit clock swings high and low. Now I understand why Marcel Licence made changes to his board to turn the buttons into voltage dividers on one pin... https://www.youtube.com/watch?v=r0af0DB1R68

**Reply by IrregularShed** [reply]
(Thank you so much for your help, I really appreciate it!)

**Comment by pschatzmann** [comment]
I create a new id so that I can handle the conflicts properly

**Comment by pschatzmann** [comment, marked as answer]
I committed some changes. You will need to update the following libraries
- AudioKit
- AudioTools (I do not register the volume functionality for setting 1 if sd is active and 7 )
